app/Console/Commands/SmokepingGenerateCommand.php

```diff
@@ -128,6 +128,7 @@ public function buildTargetsConfiguration($devices)
         $smokelist = [];
         foreach ($devices as $device) {
             $smokelist[$device->type][$device->hostname] = ['transport' => $device->transport];
```

Comment:
Maybe set like this: 
```php
$smokelist[$device->type][$device->hostname] = [
    'transport' => $device->transport,
    'displayname' => $device->displayName(),
];

```

Reply:
This entire loop could be replaced by `$devices->groupBy(['type', 'hostname'])`  Then you have full access to the device later instead of just selected fields.